### ISSUE TITLE
Undefined should be a valid part of a note title

### ISSUE THREAD

**Issue by ramarivera** [brianpetro/obsidian-smart-connections#445]
I noticed that in one or two notes, I have the word `undefined`, and it is causing validation errors when trying to calculate blocks.

```
validate_save() {
    if (!this.key)
        return false;
    if (this.key === "")
        return false;
    if (this.key.includes("undefined"))
        return false;
    return true;
}
```

This causes the following error in the console: `Invalid block, skipping save:  `

**Comment by brianpetro** [comment]
Interesting, so you must have notes with the title or headings containing "undefined" in them?

I guess I can imagine some instances why that would make sense.

I'll have to change that to `==='undefined'` rather than using the includes function. In that case, the only collisions would be with notes titled 'undefined', which should be relatively uncommon.

Alternatively, I may just switch to hashing the keys, which would allow any titles/headings with the current validation. The reason I haven't added that yet is because the unhashed version adds some visibility.

Thanks for pointing this out @ramarivera !

🌴

**Comment by ramarivera** [comment]
> Interesting, so you must have notes with the title or headings containing "undefined" in them?

Yup, the note was exactly about how to prevent getting undefined values in X,Y case 🤣 

Thanks!
